format/src/main/scala/org/scalafmt/internal/SyntaxTokens.scala

```diff
@@ -3,92 +3,119 @@ package org.scalafmt.internal
 import scala.meta._
 import scala.meta.Token
 import scala.meta.tokens.Token._
+import scala.meta.classifiers.Classifier
 
 object SyntaxTokens {
-  private implicit class Util[A <: Tree](val tree: A) extends AnyVal {
-    def findAfter(f: Token => Boolean, p: A => Tree): Option[Token] = {
+  private implicit class XtensionUtil[A <: Tree](val tree: A) extends AnyVal {
+    def find[T <: Token](implicit ev: Classifier[Token, T]): Option[T] = {
+      find0[T](tree.tokens)(ev)
+    }
+    def findAfter[T <: Token](
+        p: A => Tree
+    )(implicit ev: Classifier[Token, T]): Option[T] = {
+      val tokens = after(p)
+      find0[T](tokens)(ev)
+    }
+    def findAfter2(f: Token => Boolean, p: A => Tree): Option[Token] = {
+      val tokens = after(p)
+      tokens.find(f)
+    }
+    private def after(p: A => Tree): Tokens = {
       val end = p(tree).tokens.last
-      tree.tokens.dropWhile(_ != end).drop(1).find(f)
+      tree.tokens.dropWhile(_ != end).drop(1)
```

Comment:
Can we binary search the index here?

```scala
  def binarySearch(tokens: Tokens, token: Token): Option[Int] = {
    def loop(lo: Int, hi: Int): Option[Int] = {
      if (lo > hi) None
      else {
        val mid = lo + ((hi - lo) / 2)
        val guess = tokens(mid)
        if (guess == token) Some(mid)
        else if (guess.end < token.end) loop(mid + 1, hi)
        else loop(lo, mid - 1)
      }
    }
    loop(0, tokens.length - 1)
  }
```

Reply:
Good idea :+1:

Reply:
Tokens extends IndexedSeq. Do you think that `indexOf` is implemented with a binary search?

Reply:
IndexedSeq does not imply OrderedSeq, binary search is only possible if you know the data is ordered, which is the case for `Tokens`.

Reply:
Make sense :+1:. We should make Tokens OrderedSeq then?

Reply:
There is no OrderedSeq, it's best to use our own binary search